flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/writer/ResultPartitionWriter.java

```diff
@@ -34,12 +31,10 @@
  * The {@link ResultPartitionWriter} is the runtime API for producing results. It
  * supports two kinds of data to be sent: buffers and events.
  */
-public class ResultPartitionWriter implements EventListener<TaskEvent> {
```

Comment:
I do not understand why this class was previously implementing `EventListener`. I couldn't find any context in which it was used that way.

Reply:
a quick search reveals `NetworkEnvironment.java` calling `taskEventDispatcher.registerWriterForIncomingTaskEvents(writer.getPartitionId(), writer);` and therefore `TaskEventDispatcher` itself but there, the `EventListener` interface was actually only used in `TaskEventDispatcher#publish()` which is somewhat strange - agreed